spec/controllers/api/v3/split_registries_controller_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'rails_helper'
+
+RSpec.describe Api::V3::SplitRegistriesController, type: :controller do
+  let(:split_1) { FactoryBot.create :split, name: "one", finished_at: Time.zone.parse('2019-11-13'), registry: { all: 100 } }
+  let(:split_2) { FactoryBot.create :split, name: "two", registry: { on: 50, off: 50 } }
+  let(:split_3) { FactoryBot.create :split, name: "three_enabled", registry: { true: 99, false: 1 }, feature_gate: true }
+
+  describe "#show" do
+    before do
+      allow(ENV).to receive(:fetch).with('EXPERIENCE_SAMPLING_WEIGHT', any_args).and_return(10)
+    end
+
+    it "includes sampling weight" do
+      get :show, params: { build_timestamp: '2019-11-11T14:35:30Z' }
+
+      expect(response).to have_http_status :ok
+      expect(response_json['experience_sampling_weight']).to eq(10)
+    end
+
+    it "returns empty with no active splits on the timestamp" do
+      expect(split_1).to be_finished
+
+      get :show, params: { build_timestamp: '2019-11-14T14:35:30Z' }
+
+      expect(response).to have_http_status :ok
+      expect(response_json['splits']).to eq({})
```

Comment:
shouldn't splits two and three appear here? i think this might be caused by using `let` instead of `let!`

Reply:
ah yeah, i was relying on the expectation before executing the controller action to decide which splits I would persist for the test. Probably not the clearest for someone coming to this file for the first time. Updated using contexts.